smaug/core/network_builder.cpp

```diff
@@ -363,6 +365,25 @@ static Network* createNetworkFromProto(const GraphProto& graphProto,
         }
     }
 
+    // By now we have created all the output tensors of each operator and
+    // forwarded them to the children, but we haven't created input tensors that
+    // don't have a source.
+    for (int i = 0; i < graphProto.nodes_size(); i++) {
+        const NodeProto& node = graphProto.nodes(i);
+        Operator* op = network->getOperator(node.name());
+        for (int j = 0; j < op->getInputs().size(); j++) {
+            // If the operator has an input tensor unconstructed, create one
+            // using the information from the protos.
+            if (!op->getInputs()[j]) {
```

Comment:
What happens when two operators share an input? Won't this create that tensor more than once?

Reply:
That's really good catch! In this case, we should be using the already created tensor instead of creating a new one.